### PR TITLE
Closes #2521 remove lifecycle deprecation code for get summary records

### LINKED ISSUE #2521: General Issue: Remove lifecycle/deprecation code for `get_summary_records()`
### Background Information

Our test coverage is currently 97%. Could we increase towards 100%? Increasing test coverage of `get_summary_records()` (65%) may help.
image

### Definition of Done

~~New tests written that covers the red gap area for `get_summary_records.R`.~~

~~Use `devtools::test_coverage()` to reproduce the report~~

* Remove lifecycle code from function and any associated tests

### PR DESCRIPTION
**Pull request by jimrothstein**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by jimrothstein** [comment]
#2521 - First attempt:

Removing lifecycle deprecation code for `get_summary_records()` now allows user to use **deprecated formal arguments,**  which ** don't think you intended.**

Could you clarify, please?

**Comment by bms63** [comment]
Please remove argument from function and related code.

also good to check tests for the argument to see if those need to be excised.

I don’t think this function gets called  by user but always good to check vignettes and templates

oh and the documentation - should be cleaned up as well

thanks!!

File: tests/testthat/test-get_summary_records.R
Comment (bms63) [inline tests/testthat/test-get_summary_records.R:1]: @jimrothstein This new test you have included is not needed.

We have been letting users know for about a year that these arguments are being removed from this admiral function.  We have a pretty generous timeline for deprecating stuff.  Once we conclude the deprecation timeline we remove everything from the function and we no longer need to do anything to support it.

Unless I am missing something?

**Review by bms63 — COMMENTED** [review]
And many thanks for jumping in here!! we try to follow tidyverse deprecation guidelines, but we are more conservative.  New deprecation timelines are coming that are multi-year!!  eeckkk!!

**Comment by jimrothstein** [comment]
Removed unneeded test... understood!

### derive_param_exposure.R
- I need to adjust this function's documentation.  (It references the two vanished variables)

### locally, test admiral throwing errors
- will investigage

**Review by bundfussr — CHANGES_REQUESTED** [review]
Please add an item to `NEWS.md`

**Comment by bundfussr** [comment]
> ### derive_param_exposure.R
> * I need to adjust this function's documentation.  (It references the two vanished variables)

Good catch! It seems that we missed it before.

File: NEWS.md [hunk 1]
```diff
@@ -37,6 +37,8 @@ example, `">2.5 x ULN"` changed to `">2.5"` for grade 3. (#2534)
   
 - test scripts, R, and markdown files for `create_single_dose_dataset` and `occds.Rmd` updated to include a `STUDYID` column because of `get_admiral_option("subject_keys")` update above. (#2501)
 
+- In `get_summary_records()`, previously deprecated formal arguments `analysis_var` and `summary_fun` now removed from function, documentation, tests etc.
```
Comment (bms63) [inline NEWS.md:40]:
```suggestion
- In `get_summary_records()`, previously deprecated formal arguments `analysis_var` and `summary_fun` now removed from function, documentation, tests etc. (#2521)
```

**Comment by bms63** [comment]
We are close!!  I scan for the deprecated arguments in the repo and think we might need to look at `derive_param_exposure`

image

**Comment by jimrothstein** [comment]
Suggestion (for now or future)

**Replace**

 @return The input dataset with a new record added for each group (with respect to the variables
#' specified for the `by_vars` parameter). 

**With**
 @return The input dataset with a new record added for each group (as defined by `by_vars` parameter).

**Comment by jimrothstein** [comment]
@bms63 

Something I did ?   (I use linux)

image

**Comment by bms63** [comment]
> @bms63
> 
> Something I did ? (I use linux)
> 
> image

Seems to be a larger issue https://forum.posit.co/t/r-cmd-check-fails-on-macos-latest-release-r-lib-actions-setup-r-v2/195084/3

**Comment by bms63** [comment]
> Removed unneeded test... understood!
> 
> ### derive_param_exposure.R
> * I need to adjust this function's documentation.  (It references the two vanished variables)
> 
> ### locally, test admiral throwing errors
> * will investigage

Hi @jimrothstein happy to see this update if you still want to make it.  After that this is good to go!

**Comment by bms63** [comment]
> Suggestion (for now or future)
> 
> **Replace**
> 
> @return The input dataset with a new record added for each group (with respect to the variables #' specified for the `by_vars` parameter).
> 
> **With** @return The input dataset with a new record added for each group (as defined by `by_vars` parameter).

Hi @jimrothstein is this completed?  Love to get this merged in!

**Comment by jimrothstein** [comment]
@bms63 

Same phrase is used about 1/2 dozen times in Admiral,   change each ??

> > Suggestion (for now or future)
> > **Replace**
> > @return The input dataset with a new record added for each group (with respect to the variables #' specified for the `by_vars` parameter).
> > **With** @return The input dataset with a new record added for each group (as defined by `by_vars` parameter).
>

**Comment by bms63** [comment]
> @bms63
> 
> Same phrase is used about 1/2 dozen times in Admiral, change each ??
> 
> > > Suggestion (for now or future)
> > > **Replace**
> > > @return The input dataset with a new record added for each group (with respect to the variables #' specified for the `by_vars` parameter).
> > > **With** @return The input dataset with a new record added for each group (as defined by `by_vars` parameter).

So lets stay focus on the issue at hand which is the deprecation updates.   If not, then our PRs will never be completed! :) 

Can you create a new issue for this documentation update.  If you already did the minor update for this function here then that is fine!  no need to revert, but if not already completed lets add that one to the new documentation issue.

**Comment by bms63** [comment]
@jimrothstein I see it was updated here for the one function!  Thanks! Please make an issue for the other documentation updates that you found.

**Pull request merged**